### PR TITLE
Fix WASM JS Compatibility: Update App() Integration with SudokuHandler

### PR DESCRIPTION
**Pull request by teogor**
### Overview

This pull request addresses a compatibility issue with the WASM JS environment. Specifically, it resolves the problem where `viewModel()` is not fully supported and causes the `App()` function to malfunction in a WASM JS browser context.

### Reason for Change

The use of `viewModel()` in the `App()` function was causing issues in the WASM JS environment. As `viewModel()` is not fully functional in this context, the update uses `remember` to create an instance of `SudokuHandler`, which ensures that `App()` operates correctly within WASM JS browsers.

### CODE REVIEW

File: composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import androidx.compose.runtime.remember
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.window.ComposeViewport
 import kotlinx.browser.document
 
 @OptIn(ExperimentalComposeUiApi::class)
 fun main() {
     ComposeViewport(document.body!!) {
-        App()
+        // TODO viewModel() is not yet fully supported and does not work properly
```
Comment (jakepurple13) [inline composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt:11]: Hm...I don't completely agree with this since it works on other projects of mine like https://github.com/jakepurple13/Solitaire. Plus, just having remember means that in configuration changes, which can happen in Android, the whole field would refresh. Thoughts?

Comment (jakepurple13) [reply composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt:11]: Though...I guess for wasm only it's fine, hm?

Comment (teogor) [reply composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt:11]: I don't know. Perhaps you don't get errors at build time, but the website itself might not function properly (like this example: https://jakepurple13.github.io/Solitaire/). With this update, deploying again to GitHub Pages should resolve the issue.

File: kotlin-js-store/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,2820 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (jakepurple13) [inline kotlin-js-store/yarn.lock:1]: Is this file needed?

File: composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt (hunk 1 above)
Comment (jakepurple13) [reply composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt:11]: I figured it out. The `viewModel()` method causes it... However, if I do `viewModel { SudokuHandler() }` it works!

Comment (teogor) [reply composeApp/src/wasmJsMain/kotlin/com/programmersbox/sudoku/main.kt:11]: Good to know! I'll keep that in mind!

File: kotlin-js-store/yarn.lock (hunk 2 above)
Comment (teogor) [reply kotlin-js-store/yarn.lock:1]: I'm not entirely sure. I couldn't find much information about it online, but I noticed in JetBrains' codebase that when using the `wasmJs` target, they publish this file to version control.

**Comment by jakepurple13** [comment]
Closing because we found a solution.
